src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -237,26 +241,33 @@ public class PermalinkService : IPermalinkService
 
                     return await DownloadLegacyPermalink(permalinkId, cancellationToken)
                         .OnSuccessCombineWith(legacyPermalink =>
-                            _frontendService.CreateUniversalTable(legacyPermalink, cancellationToken))
+                            _frontendService.CreateTable(legacyPermalink, cancellationToken))
                         .OnSuccessCombineWith(tuple =>
                         {
                             var (legacyPermalink, _) = tuple;
                             return _permalinkCsvMetaService.GetCsvMeta(legacyPermalink, cancellationToken);
                         })
                         .OnSuccessVoid(async tuple =>
                         {
-                            var (legacyPermalink, universalTable, csvMeta) = tuple;
+                            var (legacyPermalink, table, csvMeta) = tuple;
+
+                            // To avoid the frontend processing and returning the footnotes unnecessarily,
+                            // add them to the table view model directly
+                            table = table with
+                            {
+                                Footnotes = legacyPermalink.FullTable.SubjectMeta.Footnotes
+                            };
 
                             await UploadSnapshot(permalink,
-                                legacyPermalink.FullTable.Results,
-                                csvMeta,
-                                universalTable,
-                                cancellationToken);
+                                observations: legacyPermalink.FullTable.Results,
```

Comment:
I see you've added all named parameters here, but is it necessary given all the different types?  I'm happy to start adopting this across the board if we figure it's better to do so, so just curious what the driver is here?

Reply:
I've only really started doing this because I saw others doing it more often 😆. I do see some extra benefits of using named params though.

In addition to the protection offered when params are of the same type, the other benefits to me are:

- Readability especially where the argument name doesn't match the parameter name like we see here, `observations` vs `legacyPermalink.FullTable.Results`.
- Makes refactoring easier when parameter order changes as we don't need to reorder all of the arguments.

I suppose it actually makes things less readable if the names are all identical and it also seems a bit overkill if there's only a single param.

We could chat about it at the next Dev's team meeting to see if we want to adopt it?